battleship_oo2_test.py

```diff
@@ -11,8 +11,8 @@ def __init__(self, player1: 'Player', player2: 'Player'):
         self._player1 = player1
         self._player2 = player2
         self._winner = None
-        self.shooter = self._player1
-        self.receiver = self._player2
+        self._shooter = self._player1
+        self._receiver = self._player2
 
     def play_game(self):
```

Comment:
This is the only method I can see you use from outside Game which means all other Game methods should be private.
I think there are other examples for the same ie. public methods that should be private.

Consider creating all methods as private by default and only make those methods public that you especially want to expose to the outside world.